### ISSUE TITLE
Trouble setting up development server

### ISSUE THREAD

**Issue by mattias-p** [zonemaster/zonemaster-gui#52]
I'm having trouble setting up a development server according to https://github.com/zonemaster/zonemaster-gui/tree/develop#contribution. I get the error message "Zonemaster Backend is not available" (shown above the Domain name field on /domain_check).

I also tried updating the environment files in /src/environments/ to use a local backend API endpoint (current Backend develop branch), but the results look very similar to me.

#### Network traffic

Request URL: https://zonemaster.fr/api
Request method: OPTIONS
Remote address: 192.134.4.83:443
Status code: 200
Version: HTTP/1.1
Referrer Policy: no-referrer-when-downgrade
Request headers:
```
Host: zonemaster.fr
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:66.0) Gecko/20100101 Firefox/66.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Access-Control-Request-Method: POST
Access-Control-Request-Headers: content-type
Referer: http://192.168.56.11:4200/domain_check
Origin: http://192.168.56.11:4200
Connection: keep-alive
Cache-Control: max-age=0
```

Response headers:
```
HTTP/1.1 200 OK
Date: Fri, 17 May 2019 09:49:55 GMT
Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
Content-Type: application/json
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
Transfer-Encoding: chunked
```

The response body is empty.

#### Console
```
[WDS] Disconnected! client:175
Please set the api endpoint dns-check.service.ts:17:6
Angular is running in the development mode. Call enableProdMode() to enable the production mode. core.js:3683
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://zonemaster.fr/api. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).[Learn More]
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://zonemaster.fr/api. (Reason: CORS request did not succeed).[Learn More]
Object { headers: {…}, status: 0, statusText: "Unknown Error", url: null, ok: false, name: "HttpErrorResponse", message: "Http failure response for (unknown url): 0 Unknown Error", error: error }
dns-check.service.ts:46:10
Object { headers: {…}, status: 0, statusText: "Unknown Error", url: null, ok: false, name: "HttpErrorResponse", message: "Http failure response for (unknown url): 0 Unknown Error", error: error }
footer.component.ts:32:6
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://zonemaster.fr/api. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).[Learn More]
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://zonemaster.fr/api. (Reason: CORS request did not succeed).[Learn More]
Object { headers: {…}, status: 0, statusText: "Unknown Error", url: null, ok: false, name: "HttpErrorResponse", message: "Http failure response for (unknown url): 0 Unknown Error", error: error }
dns-check.service.ts:46:10
ERROR Error: "Uncaught (in promise): HttpErrorResponse: {"headers":{"normalizedNames":{},"lazyUpdate":null,"headers":{}},"status":0,"statusText":"Unknown Error","url":null,"ok":false,"name":"HttpErrorResponse","message":"Http failure response for (unknown url): 0 Unknown Error","error":{"isTrusted":true}}"
    Angular 11
core.js:1448
    Angular 4
    RxJS 5
    Angular 14
```

**Comment by matsduf** [comment]
@mattias-p, @pamasse and @sandoche2k! This issue has no milestone set. Is this needed to be solved for the v2019.1 release? Please set milestone.

**Comment by pamasse** [comment]
Hi @mattias-p,

In the browser, there is a lot of security system that blocks malicious script to break or to steal something. One of them is the CORS which is block request that comes from a different URL.
See an explanation: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS.

To be able to enable the CORS, you have several possibilities: 
- do what we do for production => Put an Apache in front of your server that displays the GUI on '/' or sends the request to the backend on '/'
- disable this security on your browser directly on the configuration of it or with a plugin (i personally use CORS Everywhere on FF)
- modify the backend to be able to accept cors request

I add a small note in the readme to alert developer about that.
